pkg/core/dao/cacheddao.go

```diff
@@ -315,6 +316,51 @@ func (cd *Cached) DeleteStorageItem(scripthash util.Uint160, key []byte) error {
 	return nil
 }
 
+// GetStorageItemsIterator returns iterator over all storage items.
+// Function returned can be called until first error.
+func (cd *Cached) GetStorageItemsIterator(hash util.Uint160, prefix []byte) (vm.StorageIteratorFunc, error) {
+	items, err := cd.DAO.GetStorageItems(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	sort.Slice(items, func(i, j int) bool { return bytes.Compare(items[i].Key, items[j].Key) == -1 })
+
+	cache := cd.storage.getItems(hash)
+
+	var getItemFromCache vm.StorageIteratorFunc
+	keyIndex := -1
+	getItemFromCache = func() ([]byte, []byte, error) {
+		keyIndex++
+		if keyIndex >= len(cd.storage.keys[hash]) {
+			return nil, nil, errors.New("no more items")
+		}
+		k := cd.storage.keys[hash][keyIndex]
+		v := cache[k]
+		if v.State != delOp && bytes.HasPrefix([]byte(k), prefix) {
+			val := make([]byte, len(v.StorageItem.Value))
+			copy(val, v.StorageItem.Value)
+			return []byte(k), val, nil
+		}
+		return getItemFromCache()
```

Comment:
Mixing iterative and recursive styles this way is not a good idea, IMO. This is complex enough already.

Reply:
Fixed.